crates/biome_js_analyze/src/lint/nursery/use_top_level_regex.rs

```diff
@@ -0,0 +1,80 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{
+    JsInitializerClause, JsModuleItemList, JsRegexLiteralExpression, JsVariableDeclaration,
+    JsVariableDeclarator, JsVariableDeclaratorList, JsVariableStatement,
+};
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Enforce all regex literals to be declared at the top level.
+    ///
+    /// This rule is useful to avoid performance issues when using regex literals inside functions. Regex literals are parsed every time they are used to create a new RegExp object. By declaring them at the top level, this overhead can be avoided.
```

Comment:
```suggestion
    /// This rule is useful to avoid performance issues when using regex literals inside functions called many times (hot paths). Regex literals are parsed every time they are used to create a new RegExp object. By declaring them at the top level, this overhead can be avoided.
```